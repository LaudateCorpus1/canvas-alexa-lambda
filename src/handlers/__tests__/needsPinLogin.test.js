/*
 * Copyright (C) 2018 - present Instructure, Inc.
 *
 * This file is part of Canvas.
 *
 * Canvas is free software: you can redistribute it and/or modify it under
 * the terms of the GNU Affero General Public License as published by the Free
 * Software Foundation, version 3 of the License.
 *
 * Canvas is distributed in the hope that they will be useful, but WITHOUT ANY
 * WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR
 * A PARTICULAR PURPOSE. See the GNU Affero General Public License for more
 * details.
 *
 * You should have received a copy of the GNU Affero General Public License along
 * with this program. If not, see <http://www.gnu.org/licenses/>.
 */
const axios = require("axios");
const MockAdapter = require("axios-mock-adapter");
const createVirtualAlexa = require("../../test_utils/utils.js").createVirtualAlexa;

let mock;
let alexa;

beforeAll(() => {
  alexa = createVirtualAlexa({ fakeAccessToken: false });
  mock = new MockAdapter(axios);

  // Set this up so it requires a pin
  alexa.filter(requestJSON => {
    requestJSON.session.development = true;
    requestJSON.session.user.accessToken = "PIN_REFRESH_ONLY_TOKENlocalhost;fake-access-token";
  });
});

afterEach(() => {
  mock.reset();
});

afterAll(() => {
  mock.restore();
});

it("does not let you do other intents until you verify the pin", async () => {
  const result = await alexa.utter("Blackboard is better");
  expect(result).toMatchSnapshot();
});
