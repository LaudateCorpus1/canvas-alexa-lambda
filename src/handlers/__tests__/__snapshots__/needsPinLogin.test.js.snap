// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`does not let you do other intents until you verify the pin 1`] = `
SkillResponse {
  "response": Object {
    "outputSpeech": Object {
      "ssml": "<speak>You need to login with your security pin. To login say <emphasis level=\\"strong\\">ask canvas to login with pin</emphasis>, followed by your pin digits.</speak>",
      "type": "SSML",
    },
    "shouldEndSession": true,
  },
  "sessionAttributes": Object {},
  "userAgent": "ask-node/2.7.0 Node/v8.16.1",
  "version": "1.0",
}
`;
